### PR TITLE
Use TailwindCSS for manging css colors

### PR DESCRIPTION
**Pull request by rohandebsarkar**

### CODE REVIEW

**Review by pourmand1376 — COMMENTED** [review]
Are these colors the only ones that we use? 

I am thinking of adding some predefined color palettes. I wanted to know your idea.

File: tailwind.config.js [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+  darkMode: "class",
   content: ["./content/**/*.{md,html}", "./layouts/**/*.{html,js}"],
   theme: {
-    extend: {},
+    extend: {
+      colors: {
+        primary: "var(--global-text-color)",
+        secondary: "var(--global-text-color-light)",
+        background: "var(--global-bg-color)",
+        theme: "var(--global-theme-color)",
```
Comment (pourmand1376) [inline tailwind.config.js:11]: Are these colors the only ones that we use? 

I am thinking of adding some predefined color palettes. I wanted to know your idea.

Comment (rohandebsarkar) [reply tailwind.config.js:11]: At this moment, this are the only colors we use, more can be added if necessary.

I do not have any idea how multiple pallets are managed, but seems interesting.

**Comment by pourmand1376** [comment]
I added color-palette for this theme. I just had to copy `_variables.scss` into `main.scss`. Everything else is the same.

Now users can define any color theme in `data/themes/a_sample_name.yaml` and use that by setting `color_theme=a_sample_name`. 

I will add some more templates in the future.

**Comment by pourmand1376** [comment]
My last commit adds a design system colors to this website. There are 8 colors to choose. 

We can always add more design systems if we want. I explained about the need for this one in below issue:
- #42 

We can add more design systems from [here](https://materialui.co/flatuicolors).

**Comment by pourmand1376** [comment]
Also, I had an idea, right now, we only have four colors available for each mode (dark and light). We can have more! These design systems have more colors for us to use. 

We can at least have two whites and two darks and use them in different places ([+](https://coredesignsystem.com/core-elements/)]
image

**Comment by rohandebsarkar** [comment]
Good job the color systems work great!

Expanding the number of color variables will require a bit of work, therefore I suggest to do it in a different PR.

Presently the color systems mostly offer different theme color but same background and text colors. I suggest to change the background and text colors to add more diversity in the themes.

**Comment by rohandebsarkar** [comment]
@pourmand1376, I have added additional colors that adds more configuration options to the user.
There are seven colors available (primary, secondary, background, theme, header, footer, divider) for both light and dark themes.

Would like to hear your feedback and merge this pull request.

File: assets/css/_themes.scss [hunk 2]
```diff
@@ -3,47 +3,41 @@
...
-  --global-theme-color: #{$theme-light-color};
-  --global-hover-color: #{$theme-light-color-hover};
-  --global-footer-bg-color: #{$grey-color-light};
-  --global-footer-text-color: #{grey-color-dark};
-  --global-footer-link-color: #{$theme-light-color};
-  --global-distill-app-color: #{$grey-color-dark};
-  --global-divider-color: #{$grey-color};
-  --global-card-bg-color: #{$white-color};
+    --global-text-color-primary: #{$lightmode-text-color};
+    --global-text-color-secondary: #{$lightmode-text-color-light};
```
Comment (pourmand1376) [inline assets/css/_themes.scss:7]: The naming convention is hard to follow for the end user. We are not using a secondary color here. We are using a shade of primary color. It should be named something like `text-color-primary`, `text-color-primary-shade`. 

The secondary color definition is very different from what we have used. For example, they use blue as primary and green as secondary. Also, they would pick different shades of these primary and secondary colors. 

 I actually talked to a UI/UX designer and asked for her idea. She said we only need a primary color for our website now, and the secondary color is unnecessary. Using this term would cause confusion. 

So, I suggest that we change the naming to like below. Also, I think we have to unify the naming convention. 

```suggestion
    --global-text-color-primary: #{$lightmode-text-color};
    --global-text-color-primary-shade: #{$lightmode-text-color-shade};
```

File: config/_default/params.yaml [hunk 3]
```diff
@@ -19,5 +22,7 @@ showRecentPosts: true
 recentNews: 5 # number of recent news to show in home page
 showRecentNews: true
 
-social_navbar: true # show social icons on navbar
-social_index: true # show social icons on index page
+socialNavbar: true # show social icons on navbar
+socialIndex: true # show social icons on index page
+
+colorTheme: blue # choose a theme from `data/themes` folder
```
Comment (pourmand1376) [inline config/_default/params.yaml:28]: I think it is better to have a default value so that users understand how to change the color theme. 

Also, I deleted `default.yaml`. Default doesn't mean anything. I think we should name all themes according to some color.

**Comment by pourmand1376** [comment]
Overall, you changes are great. The configuration has become so great.

File: config/_default/params.yaml (hunk 3 above)
Comment (rohandebsarkar) [reply config/_default/params.yaml:28]: The only reason I was using default is because the normal themes are failing the lighthouse accessibility tests, see https://app.netlify.com/sites/academic-folio/deploys/63d3ab5551c1480008672f95

**Comment by rohandebsarkar** [comment]
I have updated the color profiles.
These colors have been taken from the [TailwindCSS color palette](https://tailwindcss.com/docs/customizing-colors#default-color-palette).

https://tailwindcss.com/docs/customizing-colors#default-color-palette
https://tailwindcolor.com/

**Comment by rohandebsarkar** [comment]
If this looks good, I will merge it.

File: config/_default/params.yaml (hunk 3 above)
Comment (pourmand1376) [reply config/_default/params.yaml:30]: It is great that you mentioned this. Let's merge this request however you like and I will deal with this accessibility score later.

**Comment by pourmand1376** [comment]
> If this looks good, I will merge it.

Everything looks good to me. I think it would be better to focus on fixing normal markdown styles

**Comment by rohandebsarkar** [comment]
Sorry, but I did not get "fixing normal markdown styles". Is something not working?

**Comment by pourmand1376** [comment]
This is for another PR. Take a look at [here](https://academic-folio.netlify.app/posts/all-markdown-features/). None of the markdown styles work. Heading do not have distinction and ...
